### PR TITLE
DbAdapter: insertIgnore now supported

### PR DESCRIPTION
**Pull request by bryanlatten**

### CODE REVIEW

File: src/Adapters/Sql/InsertIgnore.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use Zend\Db\Sql\Insert;
+
+class InsertIgnore extends Insert {
+
+    /**
+     * @var array Specification array
+     */
+    protected $specifications = [
+        self::SPECIFICATION_INSERT => 'INSERT IGNORE INTO %1$s (%2$s) VALUES (%3$s)',
+        self::SPECIFICATION_SELECT => 'INSERT IGNORE INTO %1$s %2$s %3$s',
```
Comment (be-dmitry) [inline src/Adapters/Sql/InsertIgnore.php]: Why is this called SPECIFICATION_SELECT?

File: src/Adapters/Sql/InsertIgnore.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+namespace Behance\NBD\Dbal\Adapters\Sql;
+
+use Zend\Db\Sql\Insert;
+
+class InsertIgnore extends Insert {
+
+    /**
+     * @var array Specification array
+     */
+    protected $specifications = [
```
Comment (jimdelois) [inline src/Adapters/Sql/InsertIgnore.php]: `$_specifications` ?

File: src/Adapters/Sql/InsertIgnore.php (hunk 1 above)
Comment (bryanlatten) [inline src/Adapters/Sql/InsertIgnore.php]: this is extending the Zend SQL implementation to swap INSERT for INSERT IGNORE :)

File: tests/unit/Adapters/ZendDbAdapterTest.php [hunk 3]
```diff
@@ -397,6 +397,99 @@ public function insert() {
...
+    $result->expects( $this->once() )
+      ->method( 'getGeneratedValue' )
+      ->will( $this->returnValue( $query_result ) );
+
+    $this->assertEquals( $query_result, $adapter->insertIgnore( $table, $params ) );
+
+  } // insertIgnoreRaw
+
+
+
```
Comment (mlebrun) [inline tests/unit/Adapters/ZendDbAdapterTest.php]: extra new line, no big deal

**Comment by mlebrun** [comment]
:+1:

File: .travis.yml [hunk 4]
```diff
@@ -5,7 +5,6 @@ php:
   - 7.0
   - 5.6
   - 5.5
-  - 5.4
```
Comment (jimdelois) [inline .travis.yml]: Why no test on `5.4` ?  If we're not testing on it, should we add a `>=5.5` to `composer.json` ?

Comment (bryanlatten) [inline .travis.yml]: the zend dependencies now only support 5.5+ :(

I'll update the composer for that - great point

File: src/Abstracts/DbAdapterAbstract.php [hunk 5]
```diff
@@ -253,6 +253,15 @@ public function fetchPairs( $sql, array $parameters = null, $master = false ) {
   /**
    * @param string $table
    * @param array  $data
+   *
+   * @return int|bool
+   */
+  abstract public function insertIgnore( $table, array $data );
```
Comment (jimdelois) [inline src/Abstracts/DbAdapterAbstract.php]: This fellow is completely missing from the interface.

Comment (bryanlatten) [inline src/Abstracts/DbAdapterAbstract.php]: added!

**Comment by jimdelois** [comment]
:white_check_mark:

**Pull request merged**